### PR TITLE
Allow importing custom syntaxes

### LINKED ISSUE #200: Syntax highlight vimscript
Hi, Really nice project! How would I implement syntax highlighting for vimscript? Could I add a sublime syntax highlight definition file to my project somehow or do I need to modify cobalt or syntect?

### PR DESCRIPTION
**Pull request by rdimartino**
Fixes #200

I tried to follow the conversation in that issue about Mutexes vs dependency injection, but a lot has changed since that issue was created so I wasn't sure if that was still relevant.

Also not sure if adding a syntect feature to engarde is a breaking change, or how to handle that exactly.

Thanks!

### CODE REVIEW

File: crates/engarde/src/syntax.rs [hunk 1]
```diff
@@ -19,6 +20,12 @@ impl Syntax {
         }
     }
 
+    pub fn load_custom_syntaxes(&mut self, syntaxes_path: &Path) {
```
Comment (epage) [inline crates/engarde/src/syntax.rs:23]: Raw needs this function as well so they can be used interchangeably

File: src/syntax_highlight.rs [hunk 2]
```diff
@@ -19,7 +20,14 @@ use engarde::Raw as Highlight;
...
 lazy_static! {
-    static ref HIGHLIGHT: Highlight = Highlight::new();
+    static ref HIGHLIGHT: Highlight = {
+        let mut highlight = Highlight::new();
+        let syntaxes_path = Path::new("./_syntaxes");
+        if syntaxes_path.exists() {
+            highlight.load_custom_syntaxes(syntaxes_path);
+        }
+        highlight
+    };
```
Comment (epage) [inline src/syntax_highlight.rs:30]: We're going to need to create this at runtime so we can load the correct root for `_syntaxes`.

Could you do that refactor in a separate PR?

File: tests/cmd/custom_syntax_highlighting.in/_dest/index.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
Comment (epage) [inline tests/cmd/custom_syntax_highlighting.in/_dest/index.html:1]: I'm assuming the `_dest` folder shouldn't be included in the `.in/`

**Review by epage — COMMENTED** [review]
Thanks for moving this forward!

File: crates/engarde/src/syntax.rs (hunk 1 above)
Comment (rdimartino) [reply crates/engarde/src/syntax.rs:23]: Sure! Should that just be a no-op? Or should it log a warning or something?

Comment (epage) [reply crates/engarde/src/syntax.rs:23]: Let's leave it as a no-op

File: src/syntax_highlight.rs (hunk 2 above)
Comment (rdimartino) [reply src/syntax_highlight.rs:30]: Yeah, I'm happy to help out in any way to get this landed, but I'm not quite sure what you mean. I thought `lazy_static` would initialize this at runtime
> Using this macro, it is possible to have statics that require code to be executed at runtime in order to be initialized.

I could use a little more direction for what you're asking.

Comment (epage) [reply src/syntax_highlight.rs:30]: I went ahead and did it real quick: https://github.com/cobalt-org/cobalt.rs/pull/1052

File: crates/engarde/src/syntax.rs [hunk 4]
```diff
@@ -107,4 +114,24 @@ mod test {
...
+
+    const CUSTOM_CODEBLOCK_RENDERED: &str = "<pre style=\"background-color:#2b303b;\">\n\
+          <span style=\"color:#c0c5ce;\">[[[]]]</span>\
+          <span style=\"background-color:#bf616a;color:#2b303b;\">]</span>\
+          </pre>\n";
+
+    #[test]
+    fn highlight_custom_syntax() {
+        let mut syntax = Syntax::new();
+        let path = Path::new("./tests/fixtures/");
```
Comment (epage) [inline crates/engarde/src/syntax.rs:128]: We should probably have this in a folder with a little more context as we'll likely add custom syntax support in the future as well.

File: src/syntax_highlight.rs (hunk 2 above)
Comment (epage) [reply src/syntax_highlight.rs:30]: Merged

**Review by epage — APPROVED** [review]
Looks great, thanks!

**Comment by rdimartino** [comment]
Not sure why the windows tests failed, but it looks like I forgot to skip the new test when the `syntax-highlight` feature is not enabled, like the others https://github.com/cobalt-org/cobalt.rs/blob/master/tests/cli_tests.rs#L5-L11

Sorry! Fixing now

**Comment by epage** [comment]
> Not sure why the windows tests failed, but it looks like I forgot to skip the new test when the `syntax-highlight` feature is not enabled, like the others https://github.com/cobalt-org/cobalt.rs/blob/master/tests/cli_tests.rs#L5-L11
> 
> Sorry! Fixing now

Sorry I didn't say something before, windows is generally broken.  One of our dependencies makes assumption about the build machine but isn't tested with github actions and won't build.

**Comment by epage** [comment]
Forgot to say, this has been released.
